django/contrib/postgres/indexes.py

```diff
@@ -1,8 +1,33 @@
 from __future__ import unicode_literals
 
-from django.db.models import Index
+from django.db.models.indexes import Index
 
-__all__ = ['GinIndex']
+__all__ = ['BrinIndex', 'GinIndex']
+
+
+class BrinIndex(Index):
+    suffix = 'brin'
+
+    def __init__(self, *args, **kwargs):
+        self.pages_per_range = kwargs.pop('pages_per_range', 128)
+        return super(BrinIndex, self).__init__(*args, **kwargs)
+
+    def __eq__(self, other):
+        return super(BrinIndex, self).__eq__(other) and self.pages_per_range == other.pages_per_range
+
+    def __repr__(self):
+        return "<%(name)s: %(field)s, pages_per_range=%(pages_per_range)s>" % {
+            'name': self.__class__.__name__,
+            'fields': self.fields,
+            'pages_per_range': self.pages_per_range,
+        }
+
+    def create_sql(self, model, schema_editor):
+        using = ' USING brin'
+        if self.pages_per_range is not None:
+            using += " WITH (pages_per_range = {})".format(self.pages_per_range)
+        sql = super(BrinIndex, self).create_sql(model, schema_editor, using=' USING brin')
```

Comment:
Maybe a `%(with)s` placeholder could be added to `sql_create_index`. Do you know or could you check if other databases have similar syntax?